src/SprykerEco/Zed/UnzerApi/Business/Api/Response/Converter/UnzerApiResponseConverter.php

```diff
@@ -54,7 +54,9 @@ public function convertUnzerApiGuzzleResponseToUnzerApiResponseTransfer(
     ): UnzerApiResponseTransfer {
         $responseData = $this->utilEncodingService->decodeJson($httpResponse->getResponseBody(), true) ?? [];
         $unzerApiResponseTransfer = $this->createUnzerApiResponseTransfer($isSuccessful);
-        $hasInternalError = $responseData && $responseData[static::RESPONSE_DATA_IS_ERROR_KEY];
+        $hasInternalError = $responseData &&
```

Comment:
seems like first check for `responseData` is obsolete
could be simplified to `isset($responseData[static::RESPONSE_DATA_IS_ERROR_KEY]) && $responseData[static::RESPONSE_DATA_IS_ERROR_KEY] === true;`